src/pubsub/ua_pubsub_keystorage.c

```diff
@@ -265,9 +265,11 @@ setPubSubGroupEncryptingKey(UA_Server *server, UA_NodeId PubSubGroupId, UA_UInt3
                             UA_ByteString signingKey, UA_ByteString encryptingKey,
                             UA_ByteString keyNonce) {
     UA_LOCK_ASSERT(&server->serviceMutex, 1);
+    UA_UNLOCK(&server->serviceMutex);
     UA_StatusCode retval =
-        setWriterGroupEncryptionKeys(server, PubSubGroupId, securityTokenId,
+        UA_Server_setWriterGroupEncryptionKeys(server, PubSubGroupId, securityTokenId,
```

Comment:
Continue to use `setWriterGroupEncryptionKeys`.
Then you don't have to unlock/lock.

Reply:
Thanks I have updated this to use `setWriterGroupEncryptionKeys`